### PR TITLE
Add nightly Tekton tests for knative/client

### LINKED ISSUE #1575: Enable nightly job of Tekton e2e tests for client
We've Tekton-Knative e2e tests setup for client, per last WG call meeting we decided to run these e2e  tests only as (a) Nightly check (post usual nightly job) ~and (b) Pre-release check~.

This is to track configuring respective jobs at test-infra side.

### PR DESCRIPTION
**Pull request by adrcunha**
Fixes #1575

CC @navidshaikh

### CODE REVIEW

**Comment by navidshaikh** [comment]
lgtm

Thanks!

File: ci/prow/testgrid.yaml [hunk 1]
```diff
@@ -318,6 +322,9 @@ dashboards:
   - name: nightly
     test_group_name: ci-knative-client-nightly-release
     base_options: "sort-by-name="
+  - name: tekton
```
Comment (navidshaikh) [inline ci/prow/testgrid.yaml]: Sorry, I can't locate the job at https://testgrid.knative.dev/client.
How can I look at the job/logs?
@adrcunha @chizhg

Comment (adrcunha) [reply ci/prow/testgrid.yaml]: It's been running: https://prow.knative.dev/?job=ci-knative-client-tekton

Looks like the TestGrid config wasn't updated. I'll do it now.

Comment (adrcunha) [reply ci/prow/testgrid.yaml]: It's now up: https://testgrid.knative.dev/client#tekton

Comment (navidshaikh) [reply ci/prow/testgrid.yaml]: Thanks! Looking at fixing the job failure..

**Pull request merged**